solidity/contracts/OracleAggregator.sol

```diff
@@ -72,7 +72,8 @@ contract OracleAggregator is AccessControl, SimpleOracle, IOracleAggregator {
     address _tokenB,
     bytes calldata _data
   ) external override(ITokenPriceOracle, SimpleOracle) {
-    if (_canModifySupportForPair(_tokenA, _tokenB)) {
+    OracleAssignment memory _assignment = assignedOracle(_tokenA, _tokenB);
+    if (_canModifySupportForPair(_tokenA, _tokenB, _assignment)) {
```

Comment:
I think this might get cheaper if you send the storage pointer, since it wouldn't need to copy data to memory.  Although, it will read 2 times from storage later. So not entirely sure there.